plugin/src/lib.rs

```diff
@@ -106,6 +107,15 @@ impl<'t> PartialEq for ArgTy<'t> {
     }
 }
 
+impl<'t> Hash for ArgTy<'t> {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.0.hash(state);
+        let s = format!("{:?}", self.1.node);
```

Comment:
Ok, I agree that this is not elegant at all, but it works. Are you aware of any work around? (Note that, previous to this change, the `typeargs` hash is not working as expected, and it is creating a new entry for any kind of type that is pushed, even if it is the same type)

Reply:
Agreed that it is not elegant, but wie can fix that later. What worries me more is that our `PartialEq` is either not correct w.r.t. lifetimes or violates the trait's reflexivity requirement. To fix this, we'll need to add the argument position and set `inout` to `a.pos == b.pos`.